fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java

```diff
@@ -355,16 +356,24 @@ private void createLoadingTask(Database db, BrokerPendingTaskAttachment attachme
 
                 // Generate loading task and init the plan of task
                 LoadLoadingTask task = new LoadLoadingTask(db, table, brokerDesc,
-                                                           entry.getValue(), getDeadlineMs(), execMemLimit,
-                                                           strictMode, transactionId, this);
+                        entry.getValue(), getDeadlineMs(), execMemLimit,
+                        strictMode, transactionId, this, timezone);
                 UUID uuid = UUID.randomUUID();
                 TUniqueId loadId = new TUniqueId(uuid.getMostSignificantBits(), uuid.getLeastSignificantBits());
-                task.init(loadId, attachment.getFileStatusByTable(tableId),
-                          attachment.getFileNumByTable(tableId));
-                // Add tasks into list and pool
+                task.init(loadId, attachment.getFileStatusByTable(tableId), attachment.getFileNumByTable(tableId));
                 idToTasks.put(task.getSignature(), task);
                 loadStatistic.numLoadedRowsMap.put(loadId, new AtomicLong(0));
-                Catalog.getCurrentCatalog().getLoadTaskScheduler().submit(task);
+
+                // save all related tables and rollups in transaction state
+                TransactionState txnState = Catalog.getCurrentGlobalTransactionMgr().getTransactionState(transactionId);
+                if (txnState == null) {
+                    throw new UserException("txn does not exist: " + transactionId);
+                }
+                txnState.addTableIndexes(table);
+            }
+            // submit all tasks together
+            for (LoadTask loadTask : idToTasks.values()) {
```

Comment:
There are some finished task in `idToTasks` which should not be submitted.

Reply:
Fixed